modules/commons/src/main/java/org/apache/synapse/commons/transaction/TranscationManger.java

```diff
@@ -388,12 +388,7 @@ public static Transaction getTransaction() throws Exception {
 		    if (log.isDebugEnabled()) {
 			    log.debug("getTransaction Called");
 		    }
-
-		    TransactionManager txMgr = txManagers.get().get(key);
-		    txMgr.begin();
```

Comment:
Have you run all the tests? There may be invocations getTransaction() which requires txMgr.begin(). How do you handle such cases?